### ISSUE TITLE
6 out of 36 tests failed: is that critical?

### ISSUE THREAD

**Issue by rtarnaud** [EOSIO/eos#4079]
Running the tests yields the following error:

The following tests FAILED:
	 30 - nodeos_run_test (Failed)
	 32 - distributed-transactions-test (Failed)
	 33 - restart-scenarios-test-resync (Failed)
	 34 - restart-scenarios-test-hard_replay (Failed)
	 35 - restart-scenarios-test-none (Failed)
	 36 - validate_dirty_db_test (Failed)
Errors while running CTest
Makefile:119: recipe for target 'test' failed
make: *** [test] Error 8

I am particularly concerned with test 30: it looks pretty serious. My goal is to run a node and be able to vote from the CLI tool. Should I care about these failed tests?

Thanks.

Edit: Trying to create a node from the following command fails indeed:

./nodeos -e -p eosio --plugin eosio::chain_api_plugin --plugin eosio::history_api_plugin

**Comment by wanderingbort** [comment]
are there any logged messages or outputs you can share from these failures?   You can run `ctest --output-on-failure` to get a more verbose log of the problems

**Comment by rtarnaud** [comment]
@wanderingbort 

Here are the results of the tests with verbosity:

```shell
30/36 Test #30: nodeos_run_test ......................***Failed    0.23 sec
 is not a valid path
 is not a valid path
 is not a valid path
BEGIN
TEST_OUTPUT: test_output_0.txt
SERVER: localhost
PORT: 8888
 cmd: pkill -9 keosd
 cmd: programs/eosio-launcher/eosio-launcher -k 9
 cmd: pkill -9 nodeos
Stand up cluster
 cmd: programs/eosio-launcher/eosio-launcher -p 1 -n 1 -s mesh -d 1 -i 2018-06-13T16:25:56.028 -f --p2p-plugin net --nodeos --max-transaction-time 5000 --filter-on * --p2p-max-nodes-per-host 1
 ERROR: Launcher failed to launch.
 FAILURE - launcher
 ERROR: Failed to stand up eos cluster.
Test failed.
  =================================================================
  Contents of etc/eosio/node_bios/config.ini:
  File etc/eosio/node_bios/config.ini not found.
  =================================================================
  Contents of var/lib/node_bios/stderr.txt:
  File var/lib/node_bios/stderr.txt not found.
 =================================================================
== Errors see above ==
Shut down the cluster.
 cmd: programs/eosio-launcher/eosio-launcher -k 9
 cmd: pkill -9 nodeos
Shut down the wallet.
 cmd: pkill -9 keosd

      Start 31: p2p_dawn515_test
31/36 Test #31: p2p_dawn515_test .....................   Passed    8.12 sec
      Start 32: distributed-transactions-test
32/36 Test #32: distributed-transactions-test ........***Failed    0.19 sec
 is not a valid path
 is not a valid path
 is not a valid path
 cmd: programs/eosio-launcher/eosio-launcher -k 9
 cmd: pkill -9 nodeos
 cmd: pkill -9 keosd
producing nodes: 1, non-producing nodes: 3, topology: mesh, delay between nodes launch(seconds): 2
Stand up cluster
 cmd: programs/eosio-launcher/eosio-launcher -p 1 -n 4 -s mesh -d 2 -i 2018-06-13T16:26:04.338 -f --p2p-plugin net --nodeos --max-transaction-time 5000 --filter-on * --p2p-max-nodes-per-host 4
 ERROR: Launcher failed to launch.
 ERROR: Failed to stand up eos cluster.
  =================================================================
  Contents of etc/eosio/node_bios/config.ini:
  File etc/eosio/node_bios/config.ini not found.
  =================================================================
  Contents of var/lib/node_bios/stderr.txt:
  File var/lib/node_bios/stderr.txt not found.
== Errors see above ==
Shut down the cluster and cleanup.
 cmd: programs/eosio-launcher/eosio-launcher -k 9
 cmd: pkill -9 nodeos
Shut down the wallet and cleanup.
 cmd: pkill -9 keosd

      Start 33: restart-scenarios-test-resync
33/36 Test #33: restart-scenarios-test-resync ........***Failed    0.19 sec
 is not a valid path
 is not a valid path
 is not a valid path
  File "/eos/eos/build/tests/restart-scenarios-test.py", line 89, in <module>
    errorExit("Failed to stand up eos cluster.")
 cmd: programs/eosio-launcher/eosio-launcher -k 9
 cmd: pkill -9 nodeos
 cmd: pkill -9 keosd
producing nodes: 4, topology: mesh, delay between nodes launch(seconds): 1, chain sync strategy: resync
Stand up cluster
 cmd: programs/eosio-launcher/eosio-launcher -p 4 -n 4 -s mesh -d 1 -i 2018-06-13T16:26:04.533 -f --p2p-plugin net --nodeos --max-transaction-time 5000 --filter-on * --p2p-max-nodes-per-host 4
 ERROR: Launcher failed to launch.
 ERROR: Failed to stand up eos cluster.
Test failed.
  =================================================================
  Contents of etc/eosio/node_bios/config.ini:
  File etc/eosio/node_bios/config.ini not found.
  =================================================================
  Contents of var/lib/node_bios/stderr.txt:
  File var/lib/node_bios/stderr.txt not found.
 =================================================================
== Errors see above ==
Shut down the cluster.
 cmd: programs/eosio-launcher/eosio-launcher -k 9
 cmd: pkill -9 nodeos
 cmd: pkill -9 keosd

      Start 34: restart-scenarios-test-hard_replay
34/36 Test #34: restart-scenarios-test-hard_replay ...***Failed    0.20 sec
 is not a valid path
 is not a valid path
 is not a valid path
  File "/eos/eos/build/tests/restart-scenarios-test.py", line 89, in <module>
    errorExit("Failed to stand up eos cluster.")
 cmd: programs/eosio-launcher/eosio-launcher -k 9
 cmd: pkill -9 nodeos
 cmd: pkill -9 keosd
producing nodes: 4, topology: mesh, delay between nodes launch(seconds): 1, chain sync strategy: hardReplay
Stand up cluster
 cmd: programs/eosio-launcher/eosio-launcher -p 4 -n 4 -s mesh -d 1 -i 2018-06-13T16:26:04.737 -f --p2p-plugin net --nodeos --max-transaction-time 5000 --filter-on * --p2p-max-nodes-per-host 4
 ERROR: Launcher failed to launch.
 ERROR: Failed to stand up eos cluster.
Test failed.
  =================================================================
  Contents of etc/eosio/node_bios/config.ini:
  File etc/eosio/node_bios/config.ini not found.
  =================================================================
  Contents of var/lib/node_bios/stderr.txt:
  File var/lib/node_bios/stderr.txt not found.
 =================================================================
== Errors see above ==
Shut down the cluster.
 cmd: programs/eosio-launcher/eosio-launcher -k 9
 cmd: pkill -9 nodeos
 cmd: pkill -9 keosd

      Start 35: restart-scenarios-test-none
35/36 Test #35: restart-scenarios-test-none ..........***Failed    0.19 sec
 is not a valid path
 is not a valid path
 is not a valid path
  File "/eos/eos/build/tests/restart-scenarios-test.py", line 89, in <module>
    errorExit("Failed to stand up eos cluster.")
 cmd: programs/eosio-launcher/eosio-launcher -k 9
 cmd: pkill -9 nodeos
 cmd: pkill -9 keosd
producing nodes: 4, topology: mesh, delay between nodes launch(seconds): 1, chain sync strategy: none
Stand up cluster
 cmd: programs/eosio-launcher/eosio-launcher -p 4 -n 4 -s mesh -d 1 -i 2018-06-13T16:26:04.929 -f --p2p-plugin net --nodeos --max-transaction-time 5000 --filter-on * --p2p-max-nodes-per-host 4
 ERROR: Launcher failed to launch.
 ERROR: Failed to stand up eos cluster.
Test failed.
  =================================================================
  Contents of etc/eosio/node_bios/config.ini:
  File etc/eosio/node_bios/config.ini not found.
  =================================================================
  Contents of var/lib/node_bios/stderr.txt:
  File var/lib/node_bios/stderr.txt not found.
 =================================================================
== Errors see above ==
Shut down the cluster.
 cmd: programs/eosio-launcher/eosio-launcher -k 9
 cmd: pkill -9 nodeos
 cmd: pkill -9 keosd

      Start 36: validate_dirty_db_test
36/36 Test #36: validate_dirty_db_test ...............***Failed    0.18 sec
 is not a valid path
 is not a valid path
 is not a valid path
 cmd: programs/eosio-launcher/eosio-launcher -k 9
 cmd: pkill -9 nodeos
producing nodes: 1, topology: mesh, delay between nodes launch(seconds): 1, chain sync strategy: resync
Stand up cluster
 cmd: programs/eosio-launcher/eosio-launcher -p 1 -n 1 -s mesh -d 1 -i 2018-06-13T16:26:05.117 -f --p2p-plugin net --nodeos --max-transaction-time 5000 --filter-on * --p2p-max-nodes-per-host 1
 ERROR: Launcher failed to launch.
 ERROR: Failed to stand up eos cluster.
Test failed.
  =================================================================
  Contents of etc/eosio/node_bios/config.ini:
  File etc/eosio/node_bios/config.ini not found.
  =================================================================
  Contents of var/lib/node_bios/stderr.txt:
  File var/lib/node_bios/stderr.txt not found.
== Errors see above ==
Shut down the cluster.
 cmd: programs/eosio-launcher/eosio-launcher -k 9
 cmd: pkill -9 nodeos


83% tests passed, 6 tests failed out of 36

Total Test time (real) = 591.51 sec

The following tests FAILED:
	 30 - nodeos_run_test (Failed)
	 32 - distributed-transactions-test (Failed)
	 33 - restart-scenarios-test-resync (Failed)
	 34 - restart-scenarios-test-hard_replay (Failed)
	 35 - restart-scenarios-test-none (Failed)
	 36 - validate_dirty_db_test (Failed)
Errors while running CTest
```

**Comment by cj-oci** [comment]
Hi, Can you run the launcher command and paste back its output:
programs/eosio-launcher/eosio-launcher -p 1 -n 1 -s mesh -d 1 -i 2018-06-13T16:26:05.117 -f --p2p-plugin net --nodeos '--max-transaction-time 5000 --filter-on * --p2p-max-nodes-per-host 1'

Can you then use pgrep to check if nodeos instances are running
pgrep -a nodeos

If nodeos isn't running can you run nodeos directly and paste its output:
rm -rf var/lib/node_bios/state/shared_memory.* var/lib/node_bios/blocks/*
programs/nodeos/nodeos --max-transaction-time 5000 --filter-on '*' --p2p-max-nodes-per-host 1 --config-dir etc/eosio/node_bios --data-dir var/lib/node_bios --genesis-json ./genesis.json --genesis-timestamp 2018-05-13T16:26:05.117

**Comment by rtarnaud** [comment]
First of all, I would like to mention that removing the folder:
~/.local/share/eosio/nodeos/data
allowed me to run a test node successfully with this command:
```shell
./nodeos -e -p eosio --plugin eosio::chain_api_plugin --plugin eosio::history_api_plugin
```

@cj-oci 
```shell
programs/eosio-launcher/eosio-launcher -p 1 -n 1 -s mesh -d 1 -i 2018-06-13T16:26:05.117 -f --p2p-plugin net --nodeos '--max-transaction-time 5000 --filter-on * --p2p-max-nodes-per-host 1'
```
returns
```shell
 is not a valid path
```
(Yes, there is a prepending blank space. I imagine the name of the path is empty hence the issue?)

Obviously, 
```shell
pgrep -a nodeos
```
does not return anything.

**Comment by SoYouDie** [comment]
I've got similar issue. Trying to start eos on VPS with Ubuntu 16.04 and got 
```
The following tests FAILED:
	 26 - unit_test_binaryen (Failed)
	 27 - unit_test_wavm (Failed)
	 29 - plugin_test (Failed)
	 30 - nodeos_run_test (Failed)
	 32 - distributed-transactions-test (Failed)
	 33 - restart-scenarios-test-resync (Failed)
	 34 - restart-scenarios-test-hard_replay (Failed)
	 35 - restart-scenarios-test-none (Failed)
Errors while running CTest
Makefile:61: recipe for target 'test' failed
make: *** [test] Error 8
[3]-  Exit 100                /root/opt/mongodb/bin/mongod -f /root/opt/mongodb/mongod.conf
```
And when i try to start nodeos I've got an error 
```
2728377ms thread-0   net_plugin.cpp:2154           operator()           ] Error reading message from 130.211.59.178:9876: Connection reset by peer
2728406ms thread-0   net_plugin.cpp:2154           operator()           ] Error reading message from 34.252.209.121:5556: Connection reset by peer
2728578ms thread-0   net_plugin.cpp:2156           operator()           ] Peer 0.0.0.0:9000 - 4567172 closed connection
2728578ms thread-0   net_plugin.cpp:2156           operator()           ] Peer 34.226.76.22:9876 closed connection
2728578ms thread-0   net_plugin.cpp:2154           operator()           ] Error reading message from 18.188.38.175:9876: Connection reset by peer
2728578ms thread-0   net_plugin.cpp:2154           operator()           ] Error reading message from 18.221.255.38:9876: Connection reset by peer
2728578ms thread-0   net_plugin.cpp:1052           operator()           ] Error sending to peer 185.253.188.1:19877: Bad file descriptor
```

**Comment by cj-oci** [comment]
The path programs/eosio-launcher/eosio-launcher is relative to the 'build' directory. Can cd to the build directory and follow my instructions?

**Comment by rtarnaud** [comment]
@cj-oci 
I know it is relative. The "is not a valid path" message is output by eosio-launcher, not by the shell because he cannot find the binary.

**Comment by lazaridiscom** [comment]
@rtarnaud , have you solved this? Just update to 1.1.3, and you should be fine.

**Comment by dixia** [comment]
hi @lazaridiscom this should be fixed by the version of 1.4.x. let me know if that is not the case
